### PR TITLE
Unified debugging information

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Today I realized that some members of our team aren't ready/able to use _mixed debugging_ and continue to search for (less powerful) alternatives. Possibly the problem is too cluttered documentation. This PR removes one _outdated debugging_ page and fixes misinformation provided in the other _debugging page_.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the style guides

### CODE REVIEW

File: docs/debugger/mixed-debugging.md [hunk 1]
```diff
@@ -1,48 +1,3 @@
...
-
-![NetBeans Debugger](https://user-images.githubusercontent.com/26887752/209614191-b0513635-819b-4c64-a6f9-9823b90a1513.png)
-
-and your execution shall stop on the next `.enso` line of code. This mode allows
-to debug both - the Enso code as well as Java code. The stack traces shows a
-mixture of Java and Enso stack frames by default. Right-clicking on the thread
-allows one to switch to plain Java view (with a way more stack frames) and back.
-Analyzing low level details as well as Enso developer point of view shall be
-simple with such tool.
+Desribed at [Runtime Debugging](runtime-debugging.md).
```
Comment (4e6) [inline docs/debugger/mixed-debugging.md:3]: It seems that this file does not contain anything anymore and can be removed

File: docs/debugger/runtime-debugging.md [hunk 2]
```diff
@@ -1,36 +1,113 @@
...
+
+![Breakpoint in Enso](https://github.com/enso-org/enso/assets/26887752/54ae4126-f77a-4463-9647-4dd3a5f83526)
+
+...as one can seamlessly switch to debugging on the Enso interpreter itself! One
+can place breakpoint into Java class like `PanicException.java` and continue
+debugging with `F5`:
+
+![Breakpoint in Java](https://github.com/enso-org/enso/assets/26887752/db3fbe4e-3bb3-4d4a-bb2a-b5039f716c85)
+
+Should one ever want to jump back from Java to Enso one can use the _"Pause in
```
Comment (GregoryTravis) [inline docs/debugger/runtime-debugging.md:83]: If I already have breakpoints set in Enso code, do I still need to use "Pause in GraalVM Script", or will the debugger stop at the breakpoints without it?

**Review by GregoryTravis — APPROVED** [review]

File: docs/debugger/runtime-debugging.md (hunk 2 above)
Comment (JaroslavTulach) [reply docs/debugger/runtime-debugging.md:83]: Your mileage may vary. The "Pause in GraalVM Script' is using a (checksum) heuristic to find the script code. If subsequent executions of the code can find the _same breakpoint_, then it is enough to put the breakpoint into Enso code.

If we get reproducible test cases when the breakpoint isn't hit, we can investigates what's the cause and fix it. The devil is in the detail: if we have enough details, we can improve your experience.

Comment (GregoryTravis) [reply docs/debugger/runtime-debugging.md:83]: So Pause is definitely needed if an Enso breakpoint hasn't been hit yet?

File: docs/debugger/runtime-debugging.md [hunk 3]
```diff
@@ -1,36 +1,113 @@
...
-if you try to debug a specific expression, it might be a bit tricky. However,
-the following steps should help you to skip all the irrelevant code and get to
-the code you are interested in:
+Finding the mapping of the Enso source code to the engine's Java code in
+_IntelliJ_ isn't easy. Trying to find out how to debug a specific expression is
+way more tricky than in case of _VSCode_, _IGV_ or _NetBeans_. However, if you
+really want to stick with _IntelliJ_ as your only tool, following steps may help
+you to skip all the irrelevant code and get to the code you are interested in:
 
 - To debug a method called `foo`, put a breakpoint in
```
Comment (Akirathan) [inline docs/debugger/runtime-debugging.md:115]:
```suggestion
- To get the mapping to Enso source code, evaluate the following expression in the Java debugger: `this.getRootNode().getSourceSection()`. Note that this, obviously, works only when the debugger is stopped in some Truffle node code.
- To debug a method called `foo`, put a breakpoint in
```

**Review by Akirathan — APPROVED** [review]
Thanks for elaborating the docs for the mixed debugging via the VSCode extension. By the way, the reason why I am personally hesitant to use this debugging technique is that I am not satisfied with VSCode's support for Java code (via the NetBeans extension) and with the debugging experience for Java code. For me, IntelliJ just provides a better experience. The mapping between Java code and Enso code is not that bad, and is a small price to pay for the better Java debugging experience.

**Comment by JaroslavTulach** [comment]
> with the debugging experience for Java code. For me, IntelliJ just provides a better experience.

As the one who [promotes VSCode debugging via DAP](https://github.com/enso-org/enso/pull/8344/files#diff-421574b50574cfe546e86d4b3d32d79b8b2087f2fe204f68e5cf2693af43bbe1) you are probably OK with the debugging interface VSCode provides.

JVM debugging of Enso code in VSCode has the same interface. As a bonus it offers smooth switching to debugging Java code. While the debugging of Java code itself may not be on par with other IDEs, the combo just rocks. I'd recommend you to try it from time to time. 

It is certainly on par with DAP debugging with the additional Java bonus.

File: docs/debugger/runtime-debugging.md (hunk 2 above)
Comment (JaroslavTulach) [reply docs/debugger/runtime-debugging.md:83]: Until a breakpoint is hit, the debuggee runs. That's the common behavior of all the debuggers, I think.

**Review by jdunkerley — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]

File: docs/debugger/mixed-debugging.md (hunk 1 above)
Comment (JaroslavTulach) [reply docs/debugger/mixed-debugging.md:3]: I am keeping it as a _"forward link"_.

**Pull request merged**